dokan/create.c

```diff
@@ -330,9 +330,11 @@ VOID DispatchCreate(HANDLE Handle, // This handle is not for a file. It is for
   if (origFileName)
     free(origFileName);
 
+  if (eventInfo.Status != STATUS_SUCCESS) {
```

Comment:
FYI, I have been doing more digging regarding this STATUS_TIMEOUT vs STATUS_IO_TIMEOUT issue, regarding the fact in particular that the Context pointer gets sent back to user space with STATUS_TIMEOUT but doesn't with STATUS_IO_TIMEOUT.

I believe there is a discrepancy between the status check here and the one in the driver in DokanCompleteCreate(). The driver uses the NT_SUCCESS() macro which says "Evaluates to TRUE if the return value specified by Status is a success type (0 − 0x3FFFFFFF) or an informational type (0x40000000 − 0x7FFFFFFF)". STATUS_TIMEOUT happens to be in that range while STATUS_IO_TIMEOUT is an actual error.

I think we should probably make this check use NT_SUCCESS() instead. I'm worried that there might be other discrepancies in the codebase. What do you think?